### PR TITLE
feat: Added Value::Bytes support

### LINKED ISSUE #52: Implement spanner::Value support for BYTES
Current `spanner::Value` implementation: https://github.com/googleapis/google-cloud-cpp-spanner/blob/master/google/cloud/spanner/value.h

Spanner types to be supported: https://github.com/googleapis/googleapis/blob/master/google/spanner/v1/type.proto

### PR DESCRIPTION
**Pull request by devjgm**
Fixes #52

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-spanner/257)

### CODE REVIEW

File: google/cloud/spanner/value.cc [hunk 1]
```diff
@@ -274,6 +288,17 @@ StatusOr<std::string> Value::GetValue(std::string const&,
   return pv.string_value();
 }
 
+StatusOr<Value::Bytes> Value::GetValue(Value::Bytes const&,
+                                       google::protobuf::Value const& pv,
+                                       google::spanner::v1::Type const&) {
+  if (pv.kind_case() != google::protobuf::Value::kStringValue) {
+    return Status(StatusCode::kUnknown, "missing BYTES data");
```
Comment (devbww) [inline google/cloud/spanner/value.cc]: FYI: Not covered by test.

File: google/cloud/spanner/value.cc [hunk 2]
```diff
@@ -194,6 +202,12 @@ google::protobuf::Value Value::MakeValueProto(std::string s) {
   return v;
 }
 
+google::protobuf::Value Value::MakeValueProto(Value::Bytes bytes) {
```
Comment (devbww) [inline google/cloud/spanner/value.cc]: Can we avoid copying the bytes here?

Comment (devjgm) [reply google/cloud/spanner/value.cc]: Yep, good catch. Fixed.

File: google/cloud/spanner/value.cc (hunk 1 above)
Comment (devjgm) [reply google/cloud/spanner/value.cc]: Tested now.

Comment (devbww) [reply google/cloud/spanner/value.cc]: The new test looks like it hits the decode error below, but not this "missing" case.  Perhaps the other "missing"s are not hit either?

Comment (devjgm) [reply google/cloud/spanner/value.cc]: I think other missing cases are tested, like in the test named GetBadBool. I missed this one for Bytes, but I just added a couple more tests for `Value::Bytes`, which I think brings up to matching the other value-type tests.

**Pull request merged**